### PR TITLE
feat: updated cf status persistance logic & updated events for iframe

### PR DESCRIPTION
**Pull request by chesterkmr**
## Release Notes

- **New Features**
  - Introduced user feedback for failed submissions with specific error messages.
  - Added new `CompletedScreen` and `FailedScreen` components for improved flow state management.
  
- **Bug Fixes**
  - Enhanced error handling for file uploads in the `DocumentField` component.
  
- **Improvements**
  - Streamlined event tracking mechanism for user flow events.
  - Updated navigation button styling for better alignment and responsiveness.
  - Improved error handling and context management in the `Page` component.
  
- **Removals**
  - Deprecated the old `Failed` component in favor of the new `FailedScreen`.
  - Removed the `Success` component, now replaced by `CompletedScreen`.

### CODE REVIEW

File: apps/kyb-app/src/pages/CollectionFlow/CollectionFlow.tsx [hunk 1]
```diff
@@ -118,15 +126,16 @@ export const CollectionFlow = withSessionProtected(() => {
...
+  return definition && collectionFlowData ? (
     <DynamicUI initialState={initialUIState}>
       <DynamicUI.StateManager
         initialContext={initialContext}
         workflowId="1"
         definitionType={schema?.definition.definitionType}
         extensions={schema?.definition.extensions}
         definition={definition as State}
-        config={schema?.config}
+        //@ts-ignore
```
Comment (alonp99) [inline apps/kyb-app/src/pages/CollectionFlow/CollectionFlow.tsx]: No

File: services/workflows-service/src/collection-flow/controllers/collection-flow.controller.ts [hunk 2]
```diff
@@ -144,6 +131,64 @@ export class ColectionFlowController {
...
+              },
+            },
+            arrayMergeOption: ARRAY_MERGE_OPTION.BY_ID,
+          },
+        },
+        [tokenScope.projectId],
+        tokenScope.projectId,
+      );
+
+      throw new common.InternalServerErrorException(error);
```
[the remark this answers is not in the document]
Comment (alonp99) [reply services/workflows-service/src/collection-flow/controllers/collection-flow.controller.ts]: @chesterkmr fix it please

File: services/workflows-service/src/workflow/consts/index.ts [hunk 3]
```diff
@@ -1 +1 @@
-export const WORKFLOW_TERMINAL_STATES = ['done', 'success', 'failed', 'finish'] as const;
+export const WORKFLOW_TERMINAL_STATES = ['done', 'completed', 'failed', 'finish'] as const;
```
Comment (alonp99) [inline services/workflows-service/src/workflow/consts/index.ts]: What do you mean by terminal state, shouldn't it be finalStates?

File: services/workflows-service/src/collection-flow/collection-flow.service.ts [hunk 4]
```diff
@@ -287,13 +287,18 @@ export class CollectionFlowService {
     );
   }
 
-  async getCollectionFlowContext(tokenScope: ITokenScope): Promise<DefaultContextSchema> {
+  async getCollectionFlowContext(
```
Comment (alonp99) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts:290]: you can cover the regression we had by testing the endpoint fo this function, making sure it return config, on the kyb app side, you can just unit test the override logic works I guess

File: apps/kyb-app/src/hooks/useFlowTracking/enums/index.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export const CollectionFlowEventsEnum = {
```
Comment (tomer-shvadron) [inline apps/kyb-app/src/hooks/useFlowTracking/enums/index.ts]: You can rename the enum to `CollectionFlowEvents`

File: apps/kyb-app/src/hooks/useFlowTracking/enums/index.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  USER_EXITED: 'user-exited',
+  FLOW_COMPLETED: 'flow-completed',
+  FLOW_FAILED: 'flow-failed',
+} as const;
+
+export type CollectionFlowEventsEnum =
```
Comment (tomer-shvadron) [inline apps/kyb-app/src/hooks/useFlowTracking/enums/index.ts]: To differentiate from the enum, let's rename the type to `TCollectionFlowEvents`

File: apps/kyb-app/src/pages/CollectionFlow/components/pages/CompletedScreen/CompletedScreen.tsx [hunk 7]
```diff
@@ -3,16 +3,24 @@ import { useTranslation } from 'react-i18next';
...
+export const CompletedScreen = withSessionProtected(() => {
   const { t } = useTranslation();
   const { customer } = useCustomer();
+  const { trackEvent } = useFlowTracking();
 
   const { exit, isExitAvailable } = useAppExit();
 
+  useEffect(() => {
+    trackEvent(CollectionFlowEventsEnum.FLOW_COMPLETED);
+  }, [trackEvent]);
```
Comment (tomer-shvadron) [inline apps/kyb-app/src/pages/CollectionFlow/components/pages/CompletedScreen/CompletedScreen.tsx:22]: Shouldn't it be an empty array?

File: services/workflows-service/src/collection-flow/controllers/collection-flow.controller.ts [hunk 8]
```diff
@@ -144,6 +146,64 @@ export class ColectionFlowController {
     );
   }
 
+  @common.Post('/final-submission')
+  async finalSubmission(@TokenScope() tokenScope: ITokenScope, @common.Body() body: FinishFlowDto) {
```
[the remark this answers is not in the document]
Comment (tomer-shvadron) [reply services/workflows-service/src/collection-flow/controllers/collection-flow.controller.ts:137]: @chesterkmr please fix this typo

**Review by tomer-shvadron — APPROVED** [review]

File: apps/kyb-app/src/pages/CollectionFlow/components/pages/CompletedScreen/CompletedScreen.tsx (hunk 7 above)
Comment (chesterkmr) [reply apps/kyb-app/src/pages/CollectionFlow/components/pages/CompletedScreen/CompletedScreen.tsx:22]: Satisfying linter here, it wont be fired again unless prefix is changed which is dependency of `trackEvent` method.

File: services/workflows-service/src/collection-flow/collection-flow.service.ts (hunk 4 above)
Comment (chesterkmr) [reply services/workflows-service/src/collection-flow/collection-flow.service.ts:290]: done

**Review by alonp99 — APPROVED** [review]

**Pull request merged**